File: samples/VideoPlayer/src/VideoPlayerSample.cpp
// Oak Nut include
#include <onut/Renderer.h>
#include <onut/Settings.h>
#include <onut/SpriteBatch.h>
#include <onut/Texture.h>
#include <onut/VideoPlayer.h>

OVideoPlayerRef pPlayer;
OTextureRef pVideoTexture;

void initSettings()
{
    oSettings->setGameName("Video Player Sample");
}

void init()
{
    // Initialize our video using a rendering target
    pVideoTexture = OTexture::createRenderTarget({1, 1});
    pPlayer = OVideoPlayer::createWithRenderTarget(pVideoTexture);
    pPlayer->setSource("https://download.blender.org/durian/trailer/sintel_trailer-720p.mp4");
    pPlayer->play();
}

void update()
{
    pPlayer->update();
}

void render()
{
    // Clear
    oRenderer->clear(OColorHex(0));

    oSpriteBatch->begin();
    oSpriteBatch->drawRect(pVideoTexture, ORectFit(ORectFullScreen, pVideoTexture->getSize()));
    oSpriteBatch->end();
}

void postRender()
{
}

void renderUI()
{
}

File: samplesJS/VideoPlayer/assets/scripts/VideoPlayerSample.js
var player = new VideoPlayer();

player.setSource("https://download.blender.org/durian/trailer/sintel_trailer-720p.mp4");
player.play();

function update()
{
    player.update();
}

function render()
{
    // Clear
    Renderer.clear(Color.BLACK);

    var fullScreenRect = new Rect(0, 0, Renderer.getResolution());
    var texture = player.getRenderTarget();
    var drawRect = fullScreenRect.fitSize(texture.getSize());

    SpriteBatch.begin();
    SpriteBatch.drawRect(texture, drawRect);
    SpriteBatch.end();
}
